src/tedpy/ted6000.py

```diff
@@ -95,7 +95,12 @@ def system_type(self) -> SystemType:
     def _mtu_energy(self, mtu: TedMtu) -> EnergyYield:
         """Return consumption or production information for a MTU."""
         data = self.endpoint_mtudash_results[mtu.position]["DashData"]
-        return EnergyYield(int(data["Now"]), int(data["TDY"]), int(data["MTD"]))
+        now, today, month = int(data["Now"]), int(data["TDY"]), int(data["MTD"])
+        if mtu.type == MtuType.GENERATION:
+            # Invert GEN-type MTUs
+            return EnergyYield(-now, -today, -month)
+        else:
+            return EnergyYield(now, today, month)
```

Comment:
Consider making a similar change on the TED5000 side so any generation there returns positive as well.